src/reasoning/PrologClient.cpp

```diff
@@ -39,62 +42,208 @@ uint PrologClient::Instance::Get_id() const noexcept {
+void PrologClient::delete_instance(const Instance& instance){
+    PrologQueryProxy bdgs = _pl.query("del('" + _NAMESPACE + instance._class+ "','" + _NAMESPACE + instance._class+  "',[])");
 
 }
 
+
 struct Position{
     unsigned int _x;
     unsigned int _y;
+    //unsigned int _time;
 }
 
 struct TurnCard {
     Position _position;
 }
 
-void save_action(const TurnCard& action){
+struct EqualCards {
+    TurnCard _Card1;
+    TurnCard _Card2; 
+}
+
+struct classification{
+    std::string _class;
+}
+
+struct knownCard{
+    Instance _knwon_card;
+    position _position;
+    classification _class;
+}
+struct unknownCard{
+    Instance _unknown_card;
+    postion _position;
+}
+
+struct turn{
+    Instance _turn;
+    Instance _player;
+    Instance _round;
+    uint _timeInstance;
+}
+
+struct player{
+    std::string _name;
+    uint _num; //Nao is always 1, after that is 2,3,4,5,.....
+}
+
+
+struct round{
+    Instance _round;
+    turn _current_turn;
+    uint _timeInstance
+
+}
+
+void PrologClient::save_action_TurnCard(const TurnCard& action,uint timeInstant){
     //create prolog instance of TurnCard
+    Instance turn_card("TurnOneCard","Turn_Card");
+    create_instance(turn_card);
+    //assert Timestamp_instance as property from TurnCard
+    Instance timeStamp("TimeStamp","TimeStamp" + std::to_string(timeInstant));
+    create_instance(timeStamp);
+    assert_timeStamp_property(turn_card,timeStamp);
+    //assert CardPosition as an instance and linkt it to turn_card as two separate propertys
+    Instance card_position("CardPosition","Card_position" +std::to_string(timeInstant));
+    Property x_pos("hasXCoordinate",std::to_string(TurnCard._x));
+    Porperty y_pos("hasYCoordinate",std::to_string(TurnCard._y));
+    assert_property(card_position,x_pos);
+    assert_property(card_position,y_pos);
+    assert_property(turn_card,card_position);
+
+    //So now there should be a Instance od TurnCard which has as propertys: a timestamp(instance of timestamp) and a postion(instance of CardPosition)
+    
+
 }
 
-struct ActionTyp2 {
+
+void PrologClient::save_action_turn_equal_cards(const EqualCard& action,uint timeInstant){
+    save_action_TurnCard(action._Card1, timeInstant);
+    save_action_TurnCard(action._Card2, timeInstant);
+    Instance turn_equal_cards("TurnEqualCards","Turn_Equal_Cards");
+    //assert Timestamp_instance as property from turn_equal_cards
+    Instance timeStamp("TimeStamp","TimeStamp" + std::to_string(timeInstant));
+    create_instance(timeStamp);
+    assert_timeStamp_property(turn_equal_cards,timeStamp); 
 
 }
 
-void save_action(const ActionTyp2& action){
+
+void PrologClient::save_action_turn_two_unkown_cards(const unknownCard& Card1, consta unknownCard& Card2,uint TimeInstance){
+    
+    save_action_TurnCard(Card1._unknown_card, timeInstant);
+    save_action_TurnCard(Card2._unknown_card, timeInstant);
 
 }
 
+void PrologClient::instanciate_one_unknowncard(uint i,uint j,uint count){
```

Comment:
maybe gibe the parameters the name x_position and y_position instead of i and j ;)

Reply:
done!